### DISCUSSION TITLE
LilyGo T-Display-S3-Long and AXS15231B

### DISCUSSION THREAD

**Discussion by d3mac123** [Bodmer/TFT_eSPI#2962] [General]
Hi, is there a way to make the TFT_eSPI library to work with the new board from LilyGo? It use the AXS15231B library to control the display - link for reference https://github.com/Xinyuan-LilyGO/T-Display-S3-Long

**Comment by Bodmer** [comment]
At the moment TFT_eSPI does not support QSPI interface displays or that particular controller.

**Reply by Bodmer** [reply]
I had a quick look at the code in the link you provided. It would be possible to use TFT_eSPI as the graphics engine to write to sprites, then the LilyGo functions to send the sprites to the display. This is the approach used by some library users for the LilyGo AMOLED screen which uses QSPI. See [here](https://www.youtube.com/watch?v=FuD-QmKn1uo).

**Reply by ideepuraj** [reply]
> At the moment TFT_eSPI does not support QSPI interface displays or that particular controller.

@Bodmer , Do we have any plan to add the QSPI support on esp32s3 so that we can add the driver for AXS15231B ?.  I am doing some initial work and made some progress.  Will you be able to help if I can get something working?

**Comment by fundix** [comment]
Hi, I found a working example where LiLyGo S3 works with SquareLine Studio and TFT_eSPI. I've been looking at various examples for several hours and since I don't understand this part, I can't understand what part is missing to be able to get LiLyGo S3 Long working with SquareLine Studio. The driver for LiLyGo S3 Long should be used with Sprite, but I don't know what steps are needed to make it work as LiLyGo S3.

Example for LilyGo S3
https://forum.squareline.io/t/tips-on-using-the-lilygo-t-display-touch-boards-with-squareline-studio/985

**Reply by fundix** [reply]
nice demo
https://github.com/nikthefix/Lilygo_Support_T_Display_S3_Long_TFT_eSPI_Volos-nikthefix

**Reply by bao55611** [reply]
lilygo S3 Long can work with SquareLine Studio ? how do you do that?

**Reply by fundix** [reply]
> lilygo S3 Long can work with SquareLine Studio ? how do you do that?

I'm on vacation now but I found somewhere how to add a new board to SquareLine studio and then export only UI with event function definitions, which are then written in the main project.

maybe this link
http://forum.squareline.io/t/tips-on-using-the-lilygo-t-display-touch-boards-with-squareline-studio/985
